### PR TITLE
Fix pushing of docker images with tab characters in env

### PR DESCRIPTION
**Pull request by schroedingersZombie**
Replaces tab characters in Manifest with `\t` an prevents so the error mentioned in #3910

### CODE REVIEW

**Comment by lmars** [comment]
@schroedingersZombie this needs a regression test which builds a docker image with a tab in the env and successfully pushes it to Flynn (as described in #3910).

See [DockerReceiveSuite.TestConvertWhiteouts](https://github.com/flynn/flynn/blob/master/test/test_docker_receive.go#L70-L93) for a similar test.

**Comment by schroedingersZombie** [comment]
@lmars added regression test. I hope this is sufficent.

**Comment by titanous** [comment]
Looks like this needs a gofmt:

```text
These files are not properly gofmt'd:
 - test/test_docker_receive.go

Please reformat the above files using "gofmt -s -w" and commit the result
```

File: test/test_docker_receive.go [hunk 1]
```diff
@@ -183,3 +183,24 @@ func (s *DockerReceiveSuite) TestReleaseDeleteImageLayers(t *c.C) {
...
+
+	// create app
+	client := s.controllerClient(t)
+	app := &ct.App{Name: repo}
+	t.Assert(client.CreateApp(app), c.IsNil)
+
+	// flynn docker push image
+	t.Assert(flynn(t, "/", "-a", app.Name, "docker", "push", repo), Succeeds)
+
+	// check the whiteouts are effective
```
Comment (lmars) [inline test/test_docker_receive.go]: Update this comment to describe the action being taken here.

**Comment by lmars** [comment]
Thanks!

**Pull request merged**